### PR TITLE
Adds kubernetes deployment

### PR DESCRIPTION
**Pull request by ignasi35**
related to https://github.com/playframework/play-scala-grpc-example/issues/30

### CODE REVIEW

File: conf/application.conf
Comment (octonato) [inline conf/application.conf]: the RP prefix comes from the `rp` tool. I don't think we should carry this over. It's not clear why we are prefixing it with RP and it doesn't add any extra value.

File: kubernetes/playgrpc.yml
Comment (octonato) [inline kubernetes/playgrpc.yml]: can we call this file `play-grpc.yml`?

Comment (octonato) [reply kubernetes/playgrpc.yml]: Most of the properties on this file are irrelevant, I guess. We should trim it down to the minimum necessary.

I think all RP prefixed vars can be removed for instance.

File: kubernetes/playgrpc.yml
Comment (octonato) [inline kubernetes/playgrpc.yml]: Let's just stick with 9000 and 9443. We don't even need variables in the application.conf file

File: docs/src/main/paradox/kubernetes.md
Comment (octonato) [inline docs/src/main/paradox/kubernetes.md]: I find this paragraph confusing.

**Review by octonato — DISMISSED** [review]
Looking good. I left some comments and some minor suggestions.

File: conf/application.conf
Comment (octonato) [inline conf/application.conf]: remove left-over

File: kubernetes/playgrpc.yml
Comment (octonato) [inline kubernetes/playgrpc.yml]: `targetPort` may be removed

File: conf/application.conf
Comment (octonato) [inline conf/application.conf]: use service name instead

File: .travis.yml
Comment (ignasi35) [inline .travis.yml:17]: Formatting. Nothing to see here.

File: build.sbt
Comment (ignasi35) [inline build.sbt:35]: Some unnecessary complexity

File: app/controllers/HomeController.scala
Comment (ignasi35) [inline app/controllers/HomeController.scala:17]: Removed a lot of clutter.

File: conf/application.conf
Comment (ignasi35) [inline conf/application.conf:13]: Allow hardcoded values in dev/test and override points in deployment. Also: open both 9000 and 9433 simultaneously!

File: conf/application.conf
Comment (ignasi35) [inline conf/application.conf:57]:
```
    host = ${play.server.https.address}
    host = ${?DEPLOYMENT_SERVICE_NAME}
    port = ${play.server.https.port}
```
This took a bit of trial-and-error... The key is to reuse `play.server.https.*` settings but prefer `DEPLOYMENT_SERVICE_NAME` if available.

The `override-authority` is required because of the `CN` on the fake cert.

File: test/test/HelloScalaTestSpec.scala
Comment (ignasi35) [inline test/test/HelloScalaTestSpec.scala:15]: Organize imports

File: test/test/HelloSpecs2Spec.scala
Comment (ignasi35) [inline test/test/HelloSpecs2Spec.scala:12]: Organize imports

**Comment by ignasi35** [comment]
Waiting on the release of `play-grpc`

File: deployment/kubernetes-play-scala-grpc-example.yml
Comment (octonato) [inline deployment/kubernetes-play-scala-grpc-example.yml:27]: Maybe here we could add a comment saying that the secret.key must be changed and unique for each application.

File: deployment/kubernetes-play-scala-grpc-example.yml
Comment (octonato) [inline deployment/kubernetes-play-scala-grpc-example.yml:2]: Good YAML! Minimal and to the point!

File: docs/src/main/paradox/kubernetes.md
Comment (octonato) [inline docs/src/main/paradox/kubernetes.md]: Left-over. 

s/RP_KUBERNETES_POD_NAME/DEPLOYMENT_SERVICE_NAME/

**Review by octonato — DISMISSED** [review]
Great work. 

I left some minor comments and a small change request.

File: deployment/kubernetes-play-scala-grpc-example.yml
Comment (ignasi35) [reply deployment/kubernetes-play-scala-grpc-example.yml:27]: See #41 . Once this and #40 are merged we'll extract the settings override to a `prod-application.conf` which will make documenting production values a lot easier and keep the yaml files easier to read.

File: docs/src/main/paradox/kubernetes.md
Comment (ignasi35) [reply docs/src/main/paradox/kubernetes.md]:
```suggestion
running on the same process (see the client configuration in `application.conf` using `DEPLOYMENT_SERVICE_NAME `). 
```

Comment (ignasi35) [reply docs/src/main/paradox/kubernetes.md]: @renatocaval using `s/x/y` on PR review is so 2018. Use suggestions! :-)

**Pull request merged**
